core/src/elements/ons-ripple/index.js

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2013-2015 ASIAL CORPORATION
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import util from 'ons/util';
+import BaseElement from 'ons/base-element';
+import GestureDetector from 'ons/gesture-detector';
+import AnimatorJS from './animator-js';
+import AnimatorCSS from './animator-css';
+
+var Animator = [AnimatorJS, AnimatorCSS][1];
+
+class RippleElement extends BaseElement {
+
+  createdCallback() {
+    this.classList.add('ripple');
+    if (this.hasAttribute('_compiled')) {
+      ['_wave', '_background'].forEach(e => {
+        this[e] = this.getElementsByClassName('ripple_' + e)[0];
+      });
+    } else {
+      this._compile();
+    }
+
+    this._animator = new Animator();
+    this._gestureDetector = new GestureDetector(this, {holdTimeout: 300});
+
+    ['color', 'target', 'center', 'start-radius', 'background'].forEach(e => {
+      this.attributeChangedCallback(e, null, this.getAttribute(e));
+    });
+  }
+
+
+  _compile() {
+    ['_wave', '_background'].forEach(e => {
+      this[e] = document.createElement('div');
+      this[e].classList.add('ripple_' + e);
+      this.appendChild(this[e]);
```

Comment:
Since `<ons-ripple>` can be used lots of times in the same page (especially in lists) I think we should avoid doing `appendChild` to the DOM. You could try using `document.createDocumentFragment()` instead and attach that.

I don't know how big the performance win is but I believe it's important that the initialization of this element is very performant.